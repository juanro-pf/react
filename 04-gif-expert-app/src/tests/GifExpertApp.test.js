import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas a GifExpertApp', () => {
    
    test('Snapshot', () => {
        
        const wrapper= shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();

    });
    
    test('mostrar lista de categorias', () => {
        
        const categories= ['Batman', 'Superman'];

        const wrapper= shallow(<GifExpertApp defaulCategories={ categories }/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });
    

});
