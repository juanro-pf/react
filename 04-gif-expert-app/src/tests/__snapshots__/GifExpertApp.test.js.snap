// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifExpertApp Snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Pruebas a GifExpertApp mostrar lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Batman"
      key="Batman"
    />
    <GifGrid
      category="Superman"
      key="Superman"
    />
  </ol>
</div>
`;
